### PR TITLE
Instance count based on key list

### PR DESCRIPTION
**Pull request by kpudlik**

### CODE REVIEW

File: ec2-cluster/main.tf [hunk 1]
```diff
@@ -28,15 +27,14 @@ module "nodes" {
   bucket_name          = var.bucket_name
 }
 
-module "cluster-provisioner" {
+module "cluster_provisioner" {
   source            = "./modules/cluster-provisioner"
-  provisioner_count = var.instance_count
   instance_ips      = module.nodes.instance_ips
```
Comment (kpudlik) [inline ec2-cluster/main.tf:32]: What we can do more there is to zip `instance_ips` with `instance_keys` and provide a single variable so inside `cluster_provisioner` we are sure that both have the same length.

**Review by tbekas — APPROVED** [review]

File: ec2-cluster/main.tf (hunk 1 above)
Comment (tbekas) [reply ec2-cluster/main.tf:32]: I think that `instance_ips` isn't a variable, but an output.

**Review by tbekas — APPROVED** [review]

File: ec2-cluster/main.tf (hunk 1 above)
Comment (kpudlik) [reply ec2-cluster/main.tf:32]: Yes, `instance_ips` is the output related to `count` param which is set to `length(instance_keys)` so everything is dependent.

Comment (kpudlik) [reply ec2-cluster/main.tf:32]: Maybe we should actually make output with already zipped ips with keys 🤔

**Review by kriscieslak — APPROVED** [review]

**Pull request merged**